examples/c++/bmm150.cxx

```diff
@@ -44,8 +44,25 @@ int main(int argc, char **argv)
 //! [Interesting]
 
   // Instantiate an BMM150 using default I2C parameters
-  upm::BMM150 *sensor = new upm::BMM150();
+  upm::BMM150 *sensor = NULL;
 
+  for (int i = 0; i < 4 && sensor == NULL; ++i)
+  {
+    try
+    {
+      cout << "Trying i2c address " << hex << showbase << BMM150_DEFAULT_ADDR + i << endl;
+      sensor = new upm::BMM150(BMM150_I2C_BUS, BMM150_DEFAULT_ADDR + i);
+    }
+    catch (std::exception& e)
+    {
+    }
+
```

Comment:
This logic seems incorrect...  Say the device supports 0x1, 0x2, 0x3, and 0x4 as possible I2C addresses, and the device is actually configured for 0x02.  What happens in the above loop when you succeed?  Nothing, as it will continue trying the other addresses and eventually fail (and leak the memoy of the instance that did succeed).  You probably need a "break" in there.

In general, I am opposed to this kind of magic "guess the valid resources" logic - it's bug prone and no real guarantee that the device is actually at a specified location.  Some devices can verify if the real device is present (via a chip id or something similar), and not some other device that is using the same I2C address, and some (many, most?) cannot.  There is no implied guarantee here.